### PR TITLE
april gonzalez martinez

### PR DESCRIPTION
**Pull request by april7229**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    server.get('/api/users', (req, res) =>{
+        db
+        .find()
+                .then(users => {
+                    res.json({ users });
+                })
+                .catch(error => {
+                    res.json({ error });
+                    //do something with error
+            });
```
Comment (enitchals) [inline server.js]: This set of closing punctuation -- `});` -- closes your `.catch` statement, but you also need closing punctuation (`})`) for your `server.get(/api/users'...` request.

File: server.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        // function remove(id) {
+        //     return db('users')
+        //         .where('id', Number(id))
+        //         .del();
+        // }
+});
+// pull id off of req.params;
+  // invoke proper db.method(id) passing it the id.
+  // handle the promise like above
+server.listen(port, () => console.log('Server running on port '));
```
Comment (enitchals) [inline server.js]: If you want to add the port number to the console.log statement using the variable `port`, you'll need to use a template literal with backticks.

File: server.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        
+        // get returns the user object with a specified id
+        server.get('/api/users/:id', (req, res) => {
+            //grab the id fromurl parameters
+            const id = req.params.id;
+
+            db
+                .findById(id)
+                .then(user => {
+                    res.json(users[0]);
```
Comment (enitchals) [inline server.js]: Why `users[0]` here? You have already identified a single `user` using `findById` -- just return that user and you should be set.

**Comment by april7229** [comment]
thats how they showed us how to do it on vidoei followed along vidoe


Sent from Yahoo Mail. Get the app 

    On Tuesday, May 29, 2018, 7:00:02 PM EDT, enitchals <notifications@github.com> wrote:  
 
 
@enitchals commented on this pull request.

In server.js:
> +                    res.json({ users });
+                })
+                .catch(error => {
+                    res.json({ error });
+                    //do something with error
+            });
+        
+        // get returns the user object with a specified id
+        server.get('/api/users/:id', (req, res) => {
+            //grab the id fromurl parameters
+            const id = req.params.id;
+
+            db
+                .findById(id)
+                .then(user => {
+                    res.json(users[0]);

Why users[0] here? You have already identified a single user using findById -- just return that user and you should be set.

**Comment by designerexpert** [comment]
# Comments.
* Continue committing often, but use more meaningful commit messages.
* Good job trying your hand at the stretch. At least now you can begin to see the full circle of front end and back end come together.

File: client/src/App.test.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
```
Comment (designerexpert) [inline client/src/App.test.js:1]: This file is unnecessary, please remove it.

File: server.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+const cors = require('cors');
+//must be above line
+const db = require('./data/db.js');
+
+const server = express();
+
+//middleware
+server.use(morgan('dev'));
+server.use(helmet());
+server.use(express.json());
```
Comment (designerexpert) [inline server.js:13]: Solid use of Express.json since bodyParser is deprecated.

**Pull request closed without merging**